modules/internal/DefaultRectangular.chpl

```diff
@@ -1721,13 +1725,17 @@ module DefaultRectangular {
 
     const isNative = f.styleElement(QIO_STYLE_ELEMENT_IS_NATIVE_BYTE_ORDER): bool;
 
-    proc writeSpaces(dim:int) throws {
+    inline proc rwLiteral(lit:string) throws {
+      if f.writing then f._writeLiteral(lit); else f._readLiteral(lit);
+    }
+
+    proc rwSpaces(dim:int) throws {
       for i in 1..dim {
-        f <~> new ioLiteral(" ");
+        rwLiteral(" ");
       }
     }
 
-    proc recursiveArrayWriter(in idx: rank*idxType, dim=0, in last=false) throws {
+    proc recursiveArrayReaderWriter(in idx: rank*idxType, dim=0, in last=false) throws {
```

Comment:
It looks like this function was a writing function and has been turned into both a reading and writing function.  Was that intentional?

Reply:
It was already doing both reading and writing, the name change here is an update to reflect that fact, rather than a change in functionality.

Reply:
Ah, okay.  Thanks!